src/fheroes2/maps/maps_tiles.cpp

```diff
@@ -1993,23 +1979,24 @@ StreamBase & Maps::operator>>( StreamBase & msg, Tiles & tile )
     using ObjectIcnTypeUnderlyingType = std::underlying_type_t<decltype( tile._objectIcnType )>;
     static_assert( std::is_same_v<ObjectIcnTypeUnderlyingType, uint8_t>, "Type of _objectIcnType has been changed, check the logic below" );
 
-    static_assert( LAST_SUPPORTED_FORMAT_VERSION < FORMAT_VERSION_1001_RELEASE, "Remove the logic below." );
-    if ( Game::GetVersionOfCurrentSaveFile() < FORMAT_VERSION_1001_RELEASE ) {
-        ObjectIcnTypeUnderlyingType objectIcnType = MP2::OBJ_ICN_TYPE_UNKNOWN;
-        msg >> objectIcnType;
-
-        tile._objectIcnType = static_cast<MP2::ObjectIcnType>( objectIcnType >> 2 );
+    static_assert( LAST_SUPPORTED_FORMAT_VERSION < FORMAT_VERSION_1009_RELEASE, "Remove the logic below." );
+    if ( Game::GetVersionOfCurrentSaveFile() < FORMAT_VERSION_1009_RELEASE ) {
+        static_assert( LAST_SUPPORTED_FORMAT_VERSION < FORMAT_VERSION_1001_RELEASE, "Remove the logic below." );
+        if ( Game::GetVersionOfCurrentSaveFile() < FORMAT_VERSION_1001_RELEASE ) {
+            ObjectIcnTypeUnderlyingType objectIcnType = MP2::OBJ_ICN_TYPE_UNKNOWN;
+            msg >> objectIcnType;
 
-        tile._hasObjectAnimation = ( objectIcnType & 1 ) != 0;
-        tile._isMarkedAsRoad = ( objectIcnType & 2 ) != 0;
-    }
-    else {
-        ObjectIcnTypeUnderlyingType objectIcnType = MP2::OBJ_ICN_TYPE_UNKNOWN;
-        msg >> objectIcnType;
+            tile._objectIcnType = static_cast<MP2::ObjectIcnType>( objectIcnType >> 2 );
+        }
+        else {
+            ObjectIcnTypeUnderlyingType objectIcnType = MP2::OBJ_ICN_TYPE_UNKNOWN;
+            msg >> objectIcnType;
 
-        tile._objectIcnType = static_cast<MP2::ObjectIcnType>( objectIcnType );
+            tile._objectIcnType = static_cast<MP2::ObjectIcnType>( objectIcnType );
 
-        msg >> tile._hasObjectAnimation >> tile._isMarkedAsRoad;
+            bool temp;
+            msg >> temp >> temp;
+        }
     }
 
```

Comment:
In the case if we load the `FORMAT_VERSION_1009_RELEASE` save version we should not skip loading the `_objectIcnType`:
```suggestion
    else {
        ObjectIcnTypeUnderlyingType objectIcnType = MP2::OBJ_ICN_TYPE_UNKNOWN;
        msg >> objectIcnType;

        tile._objectIcnType = static_cast<MP2::ObjectIcnType>( objectIcnType );
    }

```